File: FungeeCommandsManager/src/main/java/org/openredstone/commands/Rename.java
package org.openredstone.commands;

import net.md_5.bungee.api.CommandSender;
import net.md_5.bungee.api.connection.ProxiedPlayer;
import org.openredstone.FungeeCommandsManager;
import org.openredstone.messages.ActionMessage;
import org.openredstone.messaging.MessageProxyDispatcher;
import org.openredstone.types.Action;

public class Rename extends FunCommand {
    public Rename() {
        super("rename");
    }

    @Override
    public void execute(CommandSender sender, String[] args) {
        ActionMessage actionMessage = new ActionMessage(Action.RENAME_ITEM, ((ProxiedPlayer) sender).getUniqueId(), args);
        MessageProxyDispatcher.sendMessage(
                FungeeCommandsManager.proxy,
                FungeeCommandsManager.channel,
                FungeeCommandsManager.subChannel,
                (ProxiedPlayer) sender,
                actionMessage);
    }
}

File: FungeeCommandsManager/src/main/kotlin/org/openredstone/commands/Rename.kt
package org.openredstone.commands

import net.md_5.bungee.api.CommandSender
import net.md_5.bungee.api.connection.ProxiedPlayer
import org.openredstone.FungeeCommandsManager
import org.openredstone.messages.ActionMessage
import org.openredstone.messaging.MessageProxyDispatcher
import org.openredstone.types.Action

class Rename(private val plugin: FungeeCommandsManager) : FunCommand(plugin, "rename") {
    override fun execute(
        sender: CommandSender,
        args: Array<String>
    ) {
        val actionMessage =
            ActionMessage(Action.RENAME_ITEM, (sender as ProxiedPlayer).uniqueId, args)
        MessageProxyDispatcher.sendMessage(
            plugin.proxy,
            plugin.channel,
            plugin.subChannel,
            sender,
            actionMessage
        )
    }
}
